### ISSUE TITLE
Provide more granular controls for setting active cluster

### ISSUE THREAD

**Issue by ggnall** [MaterializeInc/materialize#15650]
### Feature request

Problem
Users want more granular controls than SET CLUSTER in order to:

- Avoid ambiguity about which cluster a specific operation should be run on.
- Create initialization scripts that point to (or set) the appropriate cluster
- Avoid errors of running operations on the default cluster

Specific issues/solutions:
- [ ] https://github.com/MaterializeInc/materialize/issues/15195
- [ ] https://github.com/MaterializeInc/materialize/issues/14573
- [ ] Allow users to set a default cluster

**Comment by chuck-alt-delete** [comment]
It would also be great to inform users when their query isn’t backed by a materialized view / index and suggest they set cluster to a cluster where such an index exists. It’s easy to forget to set cluster when entering psql and slow queries ensue.

**Comment by ggnall** [comment]
Agree! Linking https://github.com/MaterializeInc/materialize/pull/15432 as well
